transport/http/peer_test.go

```diff
@@ -102,6 +104,43 @@ func TestHTTPWithRoundRobin(t *testing.T) {
 	integrationtest.Blast(ctx, t, c)
 }
 
+func TestHTTPOnSuspect(t *testing.T) {
+	server, serverAddr := spec.NewServer(t, ":0")
+
+	client, c := spec.NewClient(t, []string{serverAddr})
+	defer client.Stop()
+
+	// Exercise OnSuspect
+	ctx := context.Background()
+	ctx, cancel1 := context.WithTimeout(ctx, 5*testtime.Millisecond)
+	defer cancel1()
+	integrationtest.Timeout(ctx, c)
+
+	// Exercise the innocence window
+	ctx = context.Background()
+	ctx, cancel2 := context.WithTimeout(ctx, 5*testtime.Millisecond)
+	defer cancel2()
+	integrationtest.Timeout(ctx, c)
+
+	// Validate that the peer remains available
+	ctx = context.Background()
+	ctx, cancel3 := context.WithTimeout(ctx, 5*testtime.Millisecond)
+	defer cancel3()
+	integrationtest.Blast(ctx, t, c)
+
+	// Induce the peer management loop to exit through its shutdown path.
+	go server.Stop()
+	ctx = context.Background()
+	ctx, cancel4 := context.WithTimeout(ctx, 5*testtime.Millisecond)
+	defer cancel4()
+	for {
+		err := integrationtest.Call(ctx, c)
+		if err != nil {
+			break
+		}
+	}
```

Comment:
can we check that the peer is no longer available here? e.g., how do we know whether this call is failing because the server is stopped but the client still tries to send it requests, vs call is failing because there are no peer available? perhaps we just need to test the error